lib/Target/X86/X86FrameLowering.cpp

```diff
@@ -538,15 +580,34 @@ MachineBasicBlock *X86FrameLowering::emitStackProbeInline(
   LoopMBB->addSuccessor(ContinueMBB);
   LoopMBB->addSuccessor(LoopMBB);
 
-  return ContinueMBB;
+  // Mark all the instructions added to the prolog as frame setup.
+  if (InProlog) {
```

Comment:
Nit: you could iterate over the basic blocks below instead of unrolling the loop.